From: Mans Rullgard <mans@mansr.com>
Date: Fri, 29 Aug 2008 00:42:02 +0000 (+0100)
Subject: OMAP: Set Beagleboard default video mode to 800x600 60Hz
X-Git-Url: http://git.mansr.com/?p=linux-omap;a=commitdiff_plain;h=17b546e6d65aee124a2b691727cdb2c1dcadaec5

OMAP: Set Beagleboard default video mode to 800x600 60Hz

This sets the default video mode on the Beagleboard to
800x600 60Hz VESA CVT 0.48M3-R, 16bpp.  This is compatible
with a wider range of monitors than the old default values.

Signed-off-by: Mans Rullgard <mans@mansr.com>
---

diff --git a/drivers/video/omap/lcd_omap3beagle.c b/drivers/video/omap/lcd_omap3beagle.c
index 9637735..f2861ec 100644
--- a/drivers/video/omap/lcd_omap3beagle.c
+++ b/drivers/video/omap/lcd_omap3beagle.c
@@ -31,10 +31,6 @@
 
 #define LCD_PANEL_ENABLE_GPIO       170
 
-#define LCD_XRES		1024	
-#define LCD_YRES 		768
-#define LCD_PIXCLOCK		64000 /* in kHz */
-
 static int omap3beagle_panel_init(struct lcd_panel *panel,
 				struct omapfb_device *fbdev)
 {
@@ -65,19 +61,19 @@ static unsigned long omap3beagle_panel_get_caps(struct lcd_panel *panel)
 struct lcd_panel omap3beagle_panel = {
 	.name		= "omap3beagle",
 	.config		= OMAP_LCDC_PANEL_TFT,
-
-	.bpp		= 24,
+	.bpp		= 16,
 	.data_lines	= 24,
-	.x_res		= LCD_XRES,
-	.y_res		= LCD_YRES,
-	.hsw		= 3,		/* hsync_len (4) - 1 */
-	.hfp		= 3,		/* right_margin (4) - 1 */
-	.hbp		= 39,		/* left_margin (40) - 1 */
-	.vsw		= 1,		/* vsync_len (2) - 1 */
-	.vfp		= 2,		/* lower_margin */
-	.vbp		= 7,		/* upper_margin (8) - 1 */
-
-	.pixel_clock	= LCD_PIXCLOCK,
+
+	/* 800 x 600 @ 60 Hz  Reduced blanking VESA CVT 0.48M3-R */
+	.x_res		= 800,
+	.y_res		= 600,
+	.hfp		= 48,
+	.hsw		= 32,
+	.hbp		= 80,
+	.vfp		= 3,
+	.vsw		= 4,
+	.vbp		= 11,
+	.pixel_clock	= 35500,
 
 	.init		= omap3beagle_panel_init,
 	.cleanup	= omap3beagle_panel_cleanup,
